content/sensu-core/2.0/reference/tokens.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Tokens"
+description: "The tokens reference guide."
+weight: 1
+version: "2.0"
+product: "Sensu Core"
+menu: 
+  sensu-core-2.0:
+    parent: reference
+---
+
+## How do tokens work?
+When a check is scheduled to be executed by an agent, it first goes through a token substitution step. Any tokens matching attribute values in the check are applied, and then the check is executed. Invalid templates or unmatched tokens will return an error, which is logged and sent to the Sensu backend message transport. Checks with token matching errors will not be executed.
+
+## New and improved tokens
+Sensu 2.0 uses the [Go template][1] package to implement token substitution. Instead of using triple colons `:::` as in [1.x token substitution][2], 2.0 token substitution uses double curly braces `{{` around tokens to be substituted.
+
+## Sensu tokens specification
+
+### Token substitution syntax
+
+Tokens are invoked by wrapping references to entity or custom attributes with double curly braces, such as `{{ .ID }}` to substitute an entity's ID value. Nested Sensu [entity attributes][3] can be accessed via dot notation (e.g. `system.arch`).
+
+- `{{ system.network.interfaces.addresses }}` would be replaced with the [entity `system.network.interfaces.address` attribute][26]
+- `{{ url }}` would be replaced with a custom attribute called `url`
+- `{{ disk.warning }}` would be replaced with a custom attribute called
+  `warning` nested inside of a JSON hash called `disk`
+
+### Token substitution default values
+
+In the event that an attribute is not provided by the [entity][3], a token's default
+value will be substituted. Token default values are separated by a pipe character (`|`), and can be used to provide a "fallback value" for entities that are missing a specified token attribute.
+
+- `{{url|https://sensu.io}}` would be replaced with a [custom  attribute][3] called `url`. If no such attribute called `url` is included in the client definition, the default (or fallback) value of `https://sensu.io` will be used to substitute the token.
```

Comment:
We actually need to use the `default` custom function here. Example: https://github.com/sensu/sensu-alpha-documentation/blob/master/06-checks-and-assets.md#token-substitution-default-values.

And s/url/.URL/